tooling/build/scripts/publisher.sh

```diff
@@ -107,12 +107,25 @@ ls -al
 echo "Publishing to S3..."
 start_time=$(date +%s)
 
+
+NUMBER_OF_CORES=$(nproc)
+echo "Number of cores: $NUMBER_OF_CORES"
+
+# Set the number of concurrent S3 sync operations
+S3_SYNC_CONCURRENCY=$(( 4 * NUMBER_OF_CORES )) # 4x is an arbitrary number that should work well for most cases
+S3_SYNC_CONCURRENCY=$(( S3_SYNC_CONCURRENCY < 20 ? 10 : S3_SYNC_CONCURRENCY )) # Minimum of 20
+S3_SYNC_CONCURRENCY=$(( S3_SYNC_CONCURRENCY > 100 ? 100 : S3_SYNC_CONCURRENCY )) # Maximum of 100 (to prevent AWS from throttling us)
```

Comment:
Each site can have at most 2 builds at a time. Also, there are multiple sites and builds. Do we know whats the throttling limit?